oracle/feed_stork.go

```diff
@@ -0,0 +1,297 @@
+	metrics.ReportFuncCall(f.svcTags)
+	doneFn := metrics.ReportFuncTiming(f.svcTags)
+	defer doneFn()
+
+	// Parse the URL
+	u, err := url.Parse(f.url)
+	if err != nil {
+		log.Fatal("Error parsing URL:", err)
+		return oracletypes.AssetPair{}, nil
+	}
+	header := http.Header{}
+	header.Add("Authorization", "Basic "+base64.StdEncoding.EncodeToString([]byte(f.header)))
+
+	dialer := websocket.DefaultDialer
+	dialer.EnableCompression = true
+
+	// Connect to the WebSocket server
+	conn, resp, err := dialer.Dial(u.String(), header)
+	if err != nil {
+		if resp != nil {
+			log.Printf("Handshake failed with status: %d\n", resp.StatusCode)
+			for k, v := range resp.Header {
+				log.Printf("%s: %v\n", k, v)
+			}
+		}
+		log.Fatal("Error connecting to WebSocket:", err)
+		return oracletypes.AssetPair{}, nil
+	}
+	defer conn.Close()
+
+	log.Println("Connected to WebSocket server:", resp.Status)
+
+	err = conn.WriteMessage(websocket.TextMessage, []byte(f.message))
+	if err != nil {
+		log.Fatal("Error writing message:", err)
+		return oracletypes.AssetPair{}, nil
+	}
+
+	var msgNeed []byte
+	count := 0
+	for count < 2 {
+		_, message, err := conn.ReadMessage()
+		if err != nil {
+			log.Println("Error reading message:", err)
+			return oracletypes.AssetPair{}, nil
+
+		}
+		msgNeed = message
+		count += 1
+	}
+
+	log.Println("Interrupt received, closing connection")
+
+	var msgResp messageResponse
+	if err = json.Unmarshal(msgNeed, &msgResp); err != nil {
+		return oracletypes.AssetPair{}, nil
+	}
+	assetIds := make([]string, 0)
+	for key := range msgResp.Data {
+		assetIds = append(assetIds, key)
+	}
+	assetPairs = ConvertDataToAssetPair(msgResp.Data[assetIds[0]], assetIds[0])
+
+	return assetPairs, nil
+}
+
+func (f *storkPriceFeed) PullPrice(ctx context.Context) (
+	price decimal.Decimal,
+	err error,
+) {
+	return zeroPrice, nil
+}
+
+// ConvertDataToAssetPair converts data get from websocket to list of asset pairs
+func ConvertDataToAssetPair(data Data, assetId string) (result oracletypes.AssetPair) {
+	signedPricesOfAssetPair := []*oracletypes.SignedPriceOfAssetPair{}
+	for i := range data.SignedPrices {
+		newSignedPriceAssetPair := ConvertSignedPrice(data.SignedPrices[i])
+		signedPricesOfAssetPair = append(signedPricesOfAssetPair, &newSignedPriceAssetPair)
+	}
+	result.SignedPrices = signedPricesOfAssetPair
+	result.AssetId = assetId
+
+	return result
+}
+
+// ConvertSignedPrice converts signed price to SignedPriceOfAssetPair of Stork
+func ConvertSignedPrice(signeds SignedPrice) oracletypes.SignedPriceOfAssetPair {
+	var signedPriceOfAssetPair oracletypes.SignedPriceOfAssetPair
+
+	signature := CombineSignatureToString(signeds.TimestampedSignature.Signature)
+
+	signedPriceOfAssetPair.Signature = common.Hex2Bytes(signature)
+	signedPriceOfAssetPair.PublisherKey = signeds.PublisherKey
+	signedPriceOfAssetPair.Timestamp = ConvertTimestampToSecond(signeds.TimestampedSignature.Timestamp)
```

Comment:
This code is quite fragile and assumes prices are ALWAYS signed by relayers in seconds, which may not be the case